### PR TITLE
New package: marble5-17.12.2

### PR DESCRIPTION
**Pull request by Chocimier**
There is a warning `=> WARNING: marble5-devel-17.08.3_1: libmarbledeclarative.so not found in common/shlibs!` although the file is given in shlibs.

### CODE REVIEW

File: common/shlibs [hunk 1]
```diff
@@ -1928,6 +1928,9 @@ libphobos2-ldc.so.75 ldc-runtime-1.5.0_1
 libphobos2-ldc-debug.so.75 ldc-runtime-1.5.0_1
 libmarblewidget.so.19 marble-4.14.3_2
 libastro.so.1 marble-4.14.3_2
+libmarblewidget-qt5.so.28 marble5-17.08.3_1
+libastro.so.1 marble5-17.08.3_1
+libmarbledeclarative.so marble5-17.08.3_1
```
Comment (Gottox) [inline common/shlibs]: This tells xbps-src, that there's a ```libmarbledeclarative.so``` so in ```marble5-17.08.3_1``` ...

File: srcpkgs/marble5/template [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+replaces="libmarble>=0"
+distfiles="http://download.kde.org/stable/applications/${version}/src/${pkgname%5}-${version}.tar.xz"
+checksum=c27f8b33ad0f1cc36c5af4f23352f02ab8de4a18e8a448f8ef80b41756f2bf62
+
+marble5-devel_package() {
+	short_desc+=" - development files"
+	depends="${makedepends} ${sourcepkg}>=${version}_${revision}"
+	pkg_install() {
+		vmove usr/include
+		vmove "usr/lib/*.so"
```
Comment (Gottox) [inline srcpkgs/marble5/template]: ... but here you're moving this file to the marble5-devel subpackage.

**Comment by pullmoll** [comment]
> I'd like to get some feedback from @chneukirchen and @pullmoll here: Do you think this .so should be versioned?

We had a similar case some time ago. Without a versioned so you can't really split library and -devel. A patch could add some so version number to CMakeLists.txt. The syntax is
`set_target_properties(mylib PROPERTIES SOVERSION 1.0.0)`

**Comment by ebfe** [comment]
Please check for file conflicts with marble.

Having libastro.so.1 for marble and marble5 in common/shlibs is not going to work. Lucky enough nothing links against it and it could be dropped. Same for the unversioned .so. As long as no other package needs to link to it there is no need for a common/shlibs entry.

**Comment by pullmoll** [comment]
We can choose an arbitrary solib version to not conflict with marble. If we don't need a -devel package at all, then by all means removed it from the template :)

**Comment by ebfe** [comment]
As long as nothing wants to link against marble5s version of libastro i think this is fine:

```
diff --git a/common/shlibs b/common/shlibs
index dbd696ebc1..fc8b1c3a23 100644
--- a/common/shlibs
+++ b/common/shlibs
@@ -1929,8 +1929,6 @@ libphobos2-ldc-debug.so.75 ldc-runtime-1.5.0_1
 libmarblewidget.so.19 marble-4.14.3_2
 libastro.so.1 marble-4.14.3_2
 libmarblewidget-qt5.so.28 marble5-17.08.3_1
-libastro.so.1 marble5-17.08.3_1
-libmarbledeclarative.so marble5-17.08.3_1
 libparrot.so.6.9.0 parrot-6.9.0_1
 libkipi.so.11 libkipi-4.14.3_1
 libasr.so.0 libasr-1.0.0_1
diff --git a/srcpkgs/marble5/template b/srcpkgs/marble5/template
index a9d67b9e67..049870361d 100644
--- a/srcpkgs/marble5/template
+++ b/srcpkgs/marble5/template
@@ -13,16 +13,18 @@ makedepends="knewstuff-devel kparts-devel krunner-devel libressl-devel
 phonon-devel plasma-framework-devel qt5-devel qt5-location-devel
 qt5-plugin-mysql qt5-plugin-odbc qt5-plugin-pgsql qt5-plugin-sqlite
 qt5-plugin-tds qt5-serialport-devel qt5-webkit-devel"
-replaces="libmarble>=0"
 distfiles="http://download.kde.org/stable/applications/${version}/src/${pkgname%5}-${version}.tar.xz"
 checksum=c27f8b33ad0f1cc36c5af4f23352f02ab8de4a18e8a448f8ef80b41756f2bf62
 
+conflicts="marble"
+
 marble5-devel_package() {
 	short_desc+=" - development files"
 	depends="${makedepends} ${sourcepkg}>=${version}_${revision}"
 	pkg_install() {
 		vmove usr/include
-		vmove "usr/lib/*.so"
+		vmove usr/lib/libmarblewidget-qt5.so
+		vmove usr/lib/libastro.so
 		vmove usr/lib/cmake
 	}
 }
 ```

**Comment by Vaelatern** [comment]
@Gottox ?

**Comment by Gottox** [comment]
https://travis-ci.org/voidlinux/void-packages/jobs/307088870#L7722-L7726

**Comment by ebfe** [comment]
ac8f116bd70a3de6f7df53dc336c659d135dc77e

**Pull request closed without merging**